pkg/controller/machine-set-boot-image/platform_helpers.go

```diff
@@ -192,9 +195,71 @@ func reconcileLibvirt(machineSet *machinev1beta1.MachineSet, _ *corev1.ConfigMap
 	return false, nil, nil
 }
 
-func reconcileVSphere(machineSet *machinev1beta1.MachineSet, _ *corev1.ConfigMap, arch string) (patchRequired bool, newMachineSet *machinev1beta1.MachineSet, err error) {
-	klog.Infof("Skipping machineset %s, unsupported platform type VSphere with %s arch", machineSet.Name, arch)
-	return false, nil, nil
+func reconcileVSphere(machineSet *machinev1beta1.MachineSet, infra *osconfigv1.Infrastructure, configMap *corev1.ConfigMap, arch string, secretClient clientset.Interface) (patchRequired bool, newMachineSet *machinev1beta1.MachineSet, err error) {
```

Comment:
Since vSphere has been around since 4.4 I would be checking to make sure the infrastructure spec is correctly defined before trying to use it. If it doesn't exist this should warn and return.

Reply:
we are currently doing that here:
https://github.com/openshift/machine-config-operator/blob/e44d380686aee42f784a277236dbac49b083441e/pkg/controller/machine-set-boot-image/machine_set_boot_image_controller.go#L429

Are you referring to specific fields that I should be checking for inside?

Reply:
updated